core/manager.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package core
+
+import (
+	log "github.com/ChainSafe/log15"
+
+	scale "github.com/ChainSafe/gossamer/codec"
+	tx "github.com/ChainSafe/gossamer/common/transaction"
+	"github.com/ChainSafe/gossamer/consensus/babe"
+	"github.com/ChainSafe/gossamer/core/types"
+	"github.com/ChainSafe/gossamer/p2p"
+	"github.com/ChainSafe/gossamer/runtime"
+)
+
+type Service struct {
+	rt *runtime.Runtime
+	b  *babe.Session
+
+	msgChan <-chan p2p.Message
+}
+
+func NewService(rt *runtime.Runtime, b *babe.Session, msgChan <-chan p2p.Message) *Service {
+	return &Service{
+		rt:      rt,
+		b:       b,
+		msgChan: msgChan,
+	}
+}
+
+func (s *Service) Start() <-chan error {
+	e := make(chan error)
+	go s.start(e)
+	return e
+}
+
+func (s *Service) start(e chan error) {
+	go func(msgChan <-chan p2p.Message) {
+		msg := <-msgChan
+		msgType := msg.GetType()
+		if msgType == p2p.TransactionMsgType {
+			// process tx
+		} else if msgType == p2p.BlockAnnounceMsgType {
+			// process block
+		}
```

Comment:
Please add else clause (return error)

Reply:
I changed it to a switch statement since that probably makes more sense